### PR TITLE
#38346 Upgraded to Plugin Id

### PR DESCRIPTION
**Pull request by manneohrstrom**
This upgrades the plugin to use plugin id and support Core 18.26.

### CODE REVIEW

File: plugins/basic/README.md [hunk 1]
```diff
@@ -1 +1,16 @@
...
+
+- You can read more about Plugin development 
+  in our [Toolkit Developer Documentation](http://developer.shotgunsoftware.com/tk-core/bootstrap.html#developing-plugins).
+
+- The plugin needs to be built before it can be executed. You do this by 
+  executing the build tools found [here](https://github.com/shotgunsoftware/tk-core/blob/master/developer).
+
+- For more information about Toolkit, see http://support.shotgunsoftware.com/
+
+
```
Comment (fpainchaud) [inline plugins/basic/README.md]: Who is the target audience of this file?
For developers, I would include more info about `tk-config-pluginbasic` and building the plugin.
For users, I would include the procedure to install it in Maya. This procedure is currently in `shotgun.mod`.

File: plugins/basic/info.yml [hunk 2]
```diff
@@ -48,7 +48,7 @@ base_configuration:
   # path: "$TK_CONFIG_PLUGINBASIC_ROOT"
 
 # Value given to 'sgtk.bootstrap.ToolkitManager().entry_point'.
-entry_point: "basic"
+plugin_id: "basic.maya"
```
Comment (fpainchaud) [inline plugins/basic/info.yml]: A comment about the general syntax of this name would be nice.
A mention of the fact that this name is used in the `sgtk_plugin_...` package name should be made.

File: plugins/basic/python/tk_maya_basic/plugin_engine.py [hunk 3]
```diff
@@ -43,17 +43,14 @@ def bootstrap(sg_user, progress_callback, completed_callback, failed_callback):
...
-
     # Create a boostrap manager for the logged in user with the plug-in configuration data.
     toolkit_mgr = sgtk.bootstrap.ToolkitManager(sg_user)
-    toolkit_mgr.entry_point = manifest.entry_point
-    toolkit_mgr.base_configuration = manifest.base_configuration
-    toolkit_mgr.bundle_cache_fallback_paths = [bundle_cache_path]
 
-    can_bootstrap_engine_async = hasattr(toolkit_mgr, "bootstrap_engine_async")
+    # pass the manager to the manifest for basic init
+    manifest.initialize_manager(toolkit_mgr, PLUGIN_ROOT_PATH)
```
Comment (fpainchaud) [inline plugins/basic/python/tk_maya_basic/plugin_engine.py]: Encapsulating the initialization of the manager is nice.
Though, if more and more code starts to be generated into `manifest.py` by the build plugin script, we should consider a better way than a series of `write` lines.

Comment (manneohrstrom) [inline plugins/basic/python/tk_maya_basic/plugin_engine.py]: yep, totally agree! Hopefully this is ok for now! :)

File: plugins/basic/info.yml (hunk 2 above)
Comment (manneohrstrom) [inline plugins/basic/info.yml]: fixed!

File: plugins/basic/README.md (hunk 1 above)
Comment (manneohrstrom) [inline plugins/basic/README.md]: done!

**Pull request merged**